plugins/sigma.neo4j.bolt/sigma.neo4j.bolt.js

```diff
@@ -0,0 +1,248 @@
+            key;
+        
+        response.records.forEach(function (record) {
+            record._fields.forEach(function (primitive) {
+                if (!primitive) {
+                    return;
+                }
+                convertPrimitive(primitive);
+                if (('start' in primitive
+                        || 'end' in primitive
+                        || 'type' in primitive)
+                        && !sig.graph.isEdgePresent(primitive.identity)) {
+                    //graph.relationships.push(primitive);
+                    edgesMap[primitive.identity] = {
+                        id : primitive.identity,
+                        label : primitive.type,
+                        source : primitive.start,
+                        target : primitive.end,
+                        color : '#7D7C8E',
+                        neo4j_type : primitive.type,
+                        neo4j_data : primitive.properties
+                    };
+                } else if (!sig.graph.isNodePresent(primitive.identity)) {
+                    nodesMap[primitive.identity] =  {
+                        id : primitive.identity,
+                        label : primitive.identity,
+                        x : Math.random(),
+                        y : Math.random(),
+                        size : 1,
+                        color : '#000000',
+                        neo4j_labels : primitive.labels,
+                        neo4j_data : primitive.properties
+                    };
+                }
+            });
+        });
+        
+        // construct sigma nodes
+        for (key in nodesMap) {
+            graph.nodes.push(nodesMap[key]);
+        }
+        // construct sigma nodes
+        for (key in edgesMap) {
+            graph.edges.push(edgesMap[key]);
+        }
+        return graph;
+    }
+    
+    //pass draw the graph
+    var onResolve = function onResolve (response, addToGraph) {
+        var graph = driverResultToSigmaGraph(response);
+        if (!addToGraph) {//if it is invoked by queryStream method doesn't clear the graph
+           sig.graph.clear();
+        }
+        sig.graph.read(graph);
+        return sig;
+    };
+
+    /**
+     * This function connect to neo4j via bolt connection and create the sigmajs object
+     *
+     * @param {object} neo4jOpts.   Options to pass to neo4j; defaults:
+     * <b>{
+     *   endpoint : 'bolt://localhost',
+     *   auth : { scheme : 'basic', principal : '<neo4j_connection_username>', password : '<neo4j_connection_password>' },
+     *   version : 'v1'
+     * }</b>
+     * @param {object} sigOpts.     Options to pass to sigma.js, see sigma.js docs for details.
+     *
+     * @return {Driver} A neo4j Driver object
+     */
+    sigma.neo4j.connect = function connect (neo4jOpts, sigOpts) {
+    	if (arguments.length !== 2) {
+    		throw 'You must define both neo4j and sigma options';
+    	}
+        version = neo4jOpts.version || 'v1';
```

Comment:
Neo4j's javascript api provides [three ](https://github.com/neo4j/neo4j-javascript-driver)ways to connect to it, this plugin covers only two:

1. The Promise way via `sigma.neo4j.query()` method;
2. The Stream way via `sigma.neo4j.queryStream()` method.

In order to allow developers to access to the third (transactional) api the `sigma.neo4j.connect()` method returns a Driver object, so a developer can make a request to bolt transactional api:
```javascript
var driver = sigma.neo4j.connect(
  { auth : { principal : 'neo4j', password : 'andrea' } },
  { container: 'graph-container' }
);
var session = driver.session();
var tx = session.beginTransaction();
tx.run("MERGE (bob:Person {name : {nameParam} }) RETURN bob.name", { nameParam:'Bob' })
  .subscribe({
    onNext: function(record) {
      console.log(record._fields);
      },
    onCompleted: function() {
      // Completed!
      session.close();
    },
    onError: function(error) {
      console.log(error);
    }
  });

//decide if the transaction should be committed or rolled back
var success = false;

if (success) {
  tx.commit()
    .subscribe({
      onCompleted: function() {
        // Completed!
      },
      onError: function(error) {
        console.log(error);
        }
      });
  } else {
  //transaction is rolled black and nothing is created in the database
  console.log('rolled back');
  tx.rollback();
}
```

I hope this explanation helps